### PR TITLE
For #14920 - Show a Tabs Tray CFR Banner for grid view tip

### LINKED ISSUE #14920: Tabs Tray CFR Banner for grid view tip
## User Story
- As a user, I know that I can switch my default tab view to a grid view, so I could easily make this change if that is my preferred view. 

### Acceptance Criteria
- I am informed of the option to switch to tab grid view, when opening the tab tray for the first time after the grid view feature is available.
- The CFR is not intrusive to my browsing activity.
- I have the option of dismissing the CFR or tapping on the option that takes me to this particular setting. 
- I will continue to see this tip as long as I haven't dismissed or interacted with this tip. 

We can use these [UX mocks](https://app.abstract.com/projects/8afc4b50-7667-11e9-94ba-3558f8596822/branches/cfeee95f-b6fa-4579-842e-5b51f7dd4af2/commits/a3633820b169e4bc6ac37fe927c31d169dcd45d0/files/af6f466c-1a56-494d-ae53-20dd23f8a685/layers/2859F71C-CFB7-4049-8AB9-52B54EECC1FE?mode=build&selected=root-2859F71C-CFB7-4049-8AB9-52B54EECC1FE&sha=a3633820b169e4bc6ac37fe927c31d169dcd45d0) for reference but we will need strings from @betsymi

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #14920 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt [hunk 1]
```diff
@@ -258,7 +258,29 @@ class TabTrayView(
 
         adjustNewTabButtonsForNormalMode()
 
+        @Suppress("ComplexCondition")
         if (
+            view.context.settings().showGridViewInTabsSettings &&
+            view.context.settings().shouldShowGridViewBanner &&
+            view.context.settings().canShowCfr &&
+            tabs.size >= TAB_COUNT_SHOW_CFR
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt:266]: Does grid view have the same tabs size requirement?

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -92,6 +92,13 @@
     <!-- Text for the negative action button to dismiss the Close Tabs Banner. -->
     <string name="tab_tray_close_tabs_banner_negative_button_text">Dismiss</string>
 
+    <!-- Text for the banner message to tell users about our auto close feature. -->
```
Comment (ekager) [inline app/src/main/res/values/strings.xml:95]: Don't forget to change the string descriptions!

**Review by ekager — APPROVED** [review]
2 nits!

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt (hunk 1 above)
Comment (gabrielluong) [reply app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt:266]: Ah, I forgot to add that we are using the same CFR requirements as the auto-close except that we are gonna show the grid view banner first.

**Pull request merged**